src/components/TeacherViews/Onboarding/RoleSelection/RoleSelection.view.tsx

```diff
@@ -0,0 +1,92 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { OakRadioButton, OakRadioGroup } from "@oaknational/oak-components";
+import { Control, UseFormTrigger, useForm } from "react-hook-form";
+
+import FieldError from "@/components/SharedComponents/FieldError";
+import Input from "@/components/SharedComponents/Input";
+import OnboardingForm from "@/components/TeacherComponents/OnboardingForm/OnboardingForm";
+import {
+  OnboardingFormProps,
+  RoleSelectFormProps,
+  roleSelectFormSchema,
+} from "@/components/TeacherComponents/OnboardingForm/OnboardingForm.schema";
+
+const roleOptions: Record<string, string> = {
+  "teacher-training": "Training to become a teacher",
+  "teacher-trainer": "Teacher trainer",
+  "private-tutor": "Private tutor",
+  "adult-helping-child":
+    "Adult helping a child, e.g. with revision, homeschooling",
+  "mat-or-lea": "Working at multi-academy trust or local educational authority",
+  nonprofit: "Working at an educational nonprofit",
+  other: "Other",
+};
+
+const RoleSelectionView = () => {
+  const {
+    formState,
+    setValue,
+    handleSubmit,
+    clearErrors,
+    setError,
+    getValues,
+    control,
+    trigger,
+  } = useForm<RoleSelectFormProps>({
+    resolver: zodResolver(roleSelectFormSchema),
+    mode: "onBlur",
+    defaultValues: {
+      newsletterSignUp: true,
+    },
+  });
+
+  const handleChange = (role: "other" | "role", value: string) => {
+    setValue(role, value);
+    clearErrors(role);
+  };
+
+  return (
+    <OnboardingForm
+      heading="Which of the following best describes what you do?"
+      formState={formState}
+      handleSubmit={handleSubmit}
+      continueDisabled={
+        formState.errors.role !== undefined &&
+        formState.errors.other !== undefined
+      }
+      onSubmit={() =>
+        getValues().role === "other" &&
+        !getValues().other &&
+        setError("other", { message: "Please tell us what your role is" })
+      }
+      control={control as Control<OnboardingFormProps>}
+      trigger={trigger as UseFormTrigger<OnboardingFormProps>}
```

Comment:
🏄🏻 nothing draws my attention like a cast

Looking at the code `OnboardingForm` is only concerned with the newsletter sign-up field so I reckon it could be made generic with something like this:

```typescript
const OnboardingForm = <FieldValues extends { newsletterSignUp: boolean }>(props: {
  children: React.ReactNode;
  handleSubmit: UseFormHandleSubmit<FieldValues>;
  formState: UseFormStateReturn<FieldValues>;
  heading: string;
  continueDisabled: boolean;
  onSubmit?: () => void;
  control: Control<FieldValues>;
  trigger: UseFormTrigger<FieldValues>;
})
```

that would spare the casting of `control` and `trigger` leave `OnboardingForm` unconcerned with the fields for the other forms. 

but tbh it's a lot of typescript for very little gain. It's a fun exercise though.

Reply:
you're totally right to call it out! it's so unpleasant - it doesn't quite work with making it generic though as other parts of the form then complain that they're missing fields. I found a [few](https://github.com/react-hook-form/react-hook-form/issues/9287) [similar](https://github.com/react-hook-form/react-hook-form/issues/6726) sounding issues on the library with no solution 🫠

Reply:
I had a go and got it working with generics all the way up to the point I encountered that issue with `Path<T>` you mentioned.